backend/src/scheduling/background_service.rs

```diff
@@ -0,0 +1,89 @@
+use std::time::Duration;
+
+use anyhow::Result;
+use chrono::{TimeDelta, Utc};
+use protocol::{
+    devices::DeviceId,
+    tasks::{Task, TaskId},
+    time::{Milliseconds, Timespan},
+};
+use sqlx::SqlitePool;
+use tokio::{select, sync::mpsc::UnboundedReceiver, time::sleep};
+
+use crate::data_model::graph::DiscreteGraph;
+
+use super::scheduler::SchedulerAlgorithm;
+
+pub async fn background_service<F, TAlg>(
+    mut receiver: UnboundedReceiver<()>,
+    pool: SqlitePool,
+    algorithm_constructor: F,
+) where
+    F: FnOnce() -> TAlg,
+    TAlg: SchedulerAlgorithm,
+{
+    let mut algorithm = algorithm_constructor();
+    loop {
+        // Wait until we receive a message.
+        let msg = receiver.recv().await;
+        if msg.is_none() {
+            break;
+        }
+
+        let debounce = sleep(Duration::from_secs(5 * 60));
+
+        select! {
+            _ = debounce => {
+                if let Err(error) = run_algorithm(&pool, &mut algorithm).await {
+                    println!("Algorithm error!: {}", error);
+                }
+            }
+            msg = receiver.recv() => {
+                if msg.is_none() {
+                    break;
+                }
+            }
+        };
+    }
+}
+
+async fn run_algorithm<TAlg>(pool: &SqlitePool, algorithm: &mut TAlg) -> Result<()>
+where
+    TAlg: SchedulerAlgorithm,
+{
+    // TODO: Filter out tasks based on time
+    // TODO: Filter out tasks that have a started event
+    let tasks = sqlx::query!(
```

Comment:
Remember we need the effect aswell for the global scheduling.

Reply:
Will come when we have global scheduling